### PR TITLE
fix(DatePicker): panel body not centered when used in Table

### LINKED ISSUE #31289: [DatePicker] Cell texts are not centered when used in Table
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/jibenyongfa-antd4166-forked-7rpc8?file=/index.js)

### Steps to reproduce
Enter the reproduction link above and click the DatePicker component in Table.

### What is expected?
Cell texts are centered.

### What is actually happening?
Cell texts are not centered.
image

| Environment | Info |
|---|---|
| antd | 4.16.6 |
| React | 16.14.0 |
| System | macOS Catalina 10.15.4 |
| Browser | Chrome 91.0.4472.114(x86_64) |

### PR DESCRIPTION
**Pull request by Map1en**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close #31289

### 💡 Background and solution
DatePicker cell texts are not centered when used in Table

新增：
```
  &-date-panel {
    .@{picker-prefix-cls}-content {
      &:is(table) {
        text-align: center;
      }
    }
  }
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix DatePicker date cell not centered when used in Table.     |
| 🇨🇳 Chinese |    修复 DatePicke 在 Table 内使用时日期未居中的问题。      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/table/style/index.less [hunk 1]
```diff
@@ -27,10 +27,12 @@
   // https://github.com/ant-design/ant-design/issues/17611
   table {
     width: 100%;
-    text-align: left;
     border-radius: @table-border-radius-base @table-border-radius-base 0 0;
     border-collapse: separate;
     border-spacing: 0;
+    &:not(.@{ant-prefix}-picker-content) {
+      text-align: left;
+    }
```
Comment (afc163) [inline components/table/style/index.less:35]: 应该在 picker 的样式里加。

File: components/date-picker/style/panel.less [hunk 2]
```diff
@@ -550,6 +550,10 @@
     .@{picker-prefix-cls}-content {
       width: @picker-panel-cell-width * 7;
 
+      &:is(table) {
```
Comment (afc163) [inline components/date-picker/style/panel.less:553]: 兼容性不太好 https://developer.mozilla.org/en-US/docs/Web/CSS/:is

**Comment by afc163** [comment]
看一下 https://www.argos-ci.com/ant-design/ant-design/builds/1310

**Comment by AlanCutFlim** [comment]
另外使用其余日期组件(WeekPicker, MonthPicker, RangePicker, etc.)也有一样的问题
image

原先的样式被覆盖了
image


看起来都是受这行代码影响了
https://github.com/ant-design/ant-design/blob/b9390b441cf67d3441970a6f413867d9bcc7f2a2/components/table/style/index.less#L30

**Comment by Map1en** [comment]
> 看一下 https://www.argos-ci.com/ant-design/ant-design/builds/1310

已修改

**Pull request merged**